diff -Nuar -u a/gpxe/src/arch/i386/Makefile b/gpxe/src/arch/i386/Makefile
--- a/gpxe/src/arch/i386/Makefile	2017-10-11 21:12:30.000000000 +1100
+++ b/gpxe/src/arch/i386/Makefile	2017-10-12 06:08:20.160343367 +1100
@@ -67,6 +67,10 @@
 #
 CFLAGS			+= -Ui386
 
+GCC_HAS_NO_PIE = [ -z "`$(CC) -fno-PIE -x c -c /dev/null -o /dev/null 2>&1`" ]
+NO_PIE_FLAG := $(shell $(GCC_HAS_NO_PIE) && $(ECHO) '-fno-PIE')
+CFLAGS += $(NO_PIE_FLAG)
+
 # Locations of utilities
 #
 ISOLINUX_BIN	= /usr/lib/syslinux/isolinux.bin
